identity/app/form/PrivacyMapping.scala

```diff
@@ -1,51 +1,81 @@
 package form
 
-import com.gu.identity.model.User
-import idapiclient.UserUpdate
+import com.gu.identity.model.{Consent, User}
+import idapiclient.UserUpdateDTO
 import play.api.data.Forms._
+import play.api.data.JodaForms.jodaDate
 import play.api.data.Mapping
-import play.api.i18n.{MessagesApi, MessagesProvider}
+import play.api.i18n.MessagesProvider
 
 class PrivacyMapping extends UserFormMapping[PrivacyFormData] {
 
-  protected def formMapping(implicit messagesProvider: MessagesProvider): Mapping[PrivacyFormData] = mapping(
-    "receiveGnmMarketing" -> boolean,
-    "receive3rdPartyMarketing" -> boolean,
-    "allowThirdPartyProfiling" -> boolean
-  )(PrivacyFormData.apply)(PrivacyFormData.unapply)
+  private val dateTimeFormatISO8601: String = "yyyy-MM-dd'T'HH:mm:ssZZ"
```

Comment:
get rid of this and replace the usage below with `ISODateTimeFormat.dateTimeNoMillis`? I think they are the same thing.

Reply:
Indeed they are the same, however `ISODateTimeFormat.dateTimeNoMillis` returns `DateTimeFormatter` and it seems there is no way to get the underlying pattern string:
https://stackoverflow.com/questions/28949179/get-original-pattern-string-given-a-jdk-8-datetimeformatter